sklearn/ensemble/_gradient_boosting.pyx

```diff
@@ -112,25 +193,34 @@ def predict_stages(np.ndarray[object, ndim=2] estimators,
     cdef Py_ssize_t K = estimators.shape[1]
     cdef Tree tree
 
-    for i in range(n_estimators):
-        for k in range(K):
-            tree = estimators[i, k].tree_
-
-            # avoid buffer validation by casting to ndarray
-            # and get data pointer
-            # need brackets because of casting operator priority
-            _predict_regression_tree_inplace_fast(
-                <DTYPE_t*> X.data,
-                tree.nodes, tree.value,
-                scale, k, K, X.shape[0], X.shape[1],
-                <float64 *> (<np.ndarray> out).data)
-            ## out += scale * tree.predict(X).reshape((X.shape[0], 1))
+    if issparse(X):
+        if not isinstance(X, csr_matrix):
```

Comment:
I think this validation is unnecessary, seeing as all public entrypoints should already include a call to `check_array`. I'd rather not see ad-hoc error messages when `check_array` should make this consistent across the codebase.